src/loaders/zarr_utils/types.ts

```diff
@@ -0,0 +1,69 @@
+import SubscribableRequestQueue from "../../utils/SubscribableRequestQueue";
+
+export type TCZYX<T> = [T, T, T, T, T];
+export type SubscriberId = ReturnType<SubscribableRequestQueue["addSubscriber"]>;
+
+export type OMECoordinateTransformation =
+  | {
+      type: "identity";
+    }
+  | {
+      type: "translation";
+      translation: number[];
```

Comment:
Do these arrays have specific lengths? (ZYX?) Not sure if it's useful to capture in the type, just thinking.

Reply:
OME-Zarr technically allows any number of dimensions in any order. (Many of the hoops this loader has to jump through have to do with converting arbitrary dimension orderings from metadata into something consistent - see `orderByDimension` and `orderByTCZYX` below. I mostly try to type that converted, consistent representation as a fixed-length array.)